docs/ref/models/querysets.txt

```diff
@@ -1884,6 +1887,15 @@ Any keyword arguments passed to ``get_or_create()`` — *except* an optional one
 called ``defaults`` — will be used in a :meth:`get()` call. If an object is
 found, ``get_or_create()`` returns a tuple of that object and ``False``.
 
+.. warning::
+
+    This method is atomic assuming that the database enforces uniqueness of the
```

Comment:
Is it possible to add a check against model field.unique/meta unique/unique_together to the system check feature?

Reply:
I would have to be at run time as it cannot be achieved through static analysis. I guess it could make sense to raise a warning similar to what is raised when doing using `Paginator` without a total ordering

https://github.com/django/django/blob/e04209e181c99ac16ca769d115ac640015a83757/django/core/paginator.py#L121-L126

But doing this right for `get_or_create` will be way harder as `unique`, `unique_together`, and now `constraints` have to be expected. We'd also have to abort if any `constraints` are partial or contain expressions as it would be hard, or maybe even impossible, to introspect them correctly to determine if the lookup is against them.

Anyway, this should really be a ticket in itself and not part of this PR IMO.